src/Raven.Server/Documents/Replication/ReplicationLoader.cs

```diff
@@ -507,7 +513,38 @@ private void AddAndStartOutgoingReplication(ReplicationNode node, bool external)
             outgoingReplication.SuccessfulTwoWaysCommunication += OnOutgoingSendingSucceeded;
             _outgoing.TryAdd(outgoingReplication); // can't fail, this is a brand new instance
 
-            node.Url = node.Url.Trim();
+            if (node is ExternalReplication exNode)
+            {
+                try
+                {
+                    using (var requestExecutor = RequestExecutor.Create(exNode.TopologyDiscoveryUrls, exNode.Database, _server.Server.ClusterCertificateHolder.Certificate,
+                        DocumentConventions.Default))
+                    using (_server.ContextPool.AllocateOperationContext(out TransactionOperationContext ctx))
+                    {
+                        var cmd = new GetTcpInfoCommand("extrenal-replication");
+                        requestExecutor.Execute(cmd, ctx);
+                        node.Url = requestExecutor.Url;
+                        outgoingReplication.ConnectionInfo = cmd.Result;
```

Comment:
This is a very bad model for this to use. If this is external we set the `ConnectionInfo` but if internal we get it ourselves?
Just always supply this externally and decide how here.